stdlib/in_channel.mli

```diff
@@ -0,0 +1,161 @@
+  'a
+
+val create : t open_function
+(** TBA *)
+
+val with_file : (string -> (t -> 'a) -> 'a) open_function
+(** TBA *)
+
+val close : t -> unit
+(** Close the given channel.  Input functions raise a [Sys_error]
+    exception when they are applied to a closed input channel,
+    except [close_in], which does nothing when applied to an already
+    closed channel. *)
+
+val input : t -> bytes -> int -> int -> int
+(** [input ic buf pos len] reads up to [len] characters from
+    the given channel [ic], storing them in byte sequence [buf], starting at
+    character number [pos].
+    It returns the actual number of characters read, between 0 and
+    [len] (inclusive).
+    A return value of 0 means that the end of file was reached.
+    A return value between 0 and [len] exclusive means that
+    not all requested [len] characters were read, either because
+    no more characters were available at that time, or because
+    the implementation found it convenient to do a partial read;
+    [input] must be called again to read the remaining characters,
+    if desired.  (See also {!Stdlib.really_input} for reading
+    exactly [len] characters.)
+    Exception [Invalid_argument "input"] is raised if [pos] and [len]
+    do not designate a valid range of [buf]. *)
+
+val really_input : t -> bytes -> int -> int -> unit option
+(** [really_input ic buf pos len] reads [len] characters from channel [ic],
+    storing them in byte sequence [buf], starting at character number [pos].
+    Return [None] if the end of file is reached before [len]
+    characters have been read.
+    Raise [Invalid_argument "really_input"] if
+    [pos] and [len] do not designate a valid range of [buf]. *)
+
+val really_input_string : t -> int -> string option
+(** [really_input_string ic len] reads [len] characters from channel [ic]
+    and returns them in a new string.
+    Return [None] if the end of file is reached before [len]
+    characters have been read. *)
+
+val input_char : t -> char option
+(** Read one character from the given input channel.
+    Return [None] if there are no more characters to read. *)
+
+val input_byte : t -> int option
+(** Same as {!input_char}, but return the 8-bit integer representing
+    the character.
+    Return [None] if an end of file was reached. *)
+
+val input_binary_int : t -> int option
+(** Read an integer encoded in binary format (4 bytes, big-endian)
+    from the given input channel. See {!Stdlib.output_binary_int}.
+    Return [None] if an end of file was reached while reading the
+    integer. *)
+
+val input_line : t -> string option
+(** Read characters from the given input channel, until a
+    newline character is encountered. Return the string of
+    all characters read, without the newline character at the end.
+    Return [None] if the end of the file is reached
+    at the beginning of line. *)
+
+val input_lines : t -> string list
+(** TBA *)
+
+val fold_lines : ('a -> string -> 'a) -> 'a -> t -> 'a
```

Comment:
it would also be nice to have `iter_lines` (for me at least, it'd be compatible with my `Iter` library :blush: )

Reply:
Noted. I added it for now. Let's see how the discussion goes.